src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_message_goes_through_all_retries.cs

```diff
@@ -0,0 +1,176 @@
+﻿namespace NServiceBus.AcceptanceTests.Recoverability.Retries
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using Features;
+    using NServiceBus.Config;
+    using NServiceBus.Pipeline;
+    using NUnit.Framework;
+    using ScenarioDescriptors;
+
+    public class When_message_goes_through_all_retries : NServiceBusAcceptanceTest
```

Comment:
I think we can separate the concerns the following way:

Retry & Error Queue features:
* When FLR handles message: verify message processing is aborted and message is retried immediately
* When SLR handles message: verify message processing is aborted and message is retried with delay
* When ErrorQ behavior handled message: verify message processing is aborted and message is sent to errorqueue

"Batching feature":
* When message processing is aborted: verify that no outgoing message is dispatched (besides immediate dispatches)

Transactions:
* When message processing fails & transactions enabled (>= single queue): verify incoming message is rolled back. Dispatched messages are not rolled back
* When message processing fails & transactions enabled (>= multi queue): verify incoming message and dispatched messages are rolled back.
* When message processing fails & transactions disabled: verify incoming/dispatched message are not rolled back.

so that's why why I think we should scope our expectations to a specific behavior/feature and not a combination of features & configurations. Does that make sense somehow? (Note: I just dumped some thoughts on this list, this may be wrong or not behave like it's described on different transports)